### PR TITLE
[url_launcher] Add `InAppBrowserConfiguration` parameter in implementations

### PR DESCRIPTION
**Pull request by Alex-Usmanov**
Platform implementation portion of #5166. Implements `InAppBrowserConfiguration` support on Android, as well as support for `InAppBrowserConfiguration.showTitle` parameter for hiding/showing webpage title in Android Custom Tabs.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md.
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Marking as a draft for now; the sub-PRs can't proceed without the dependencies landing first. The overall pre-review is done in the main PR.

**Comment by stuartmorgan** [comment]
> it seems to have slipped through the cracks.

You only marked it as ready for review at the end of last week; it often takes more than two working days for a PR to be reviewed.

**Comment by Alex-Usmanov** [comment]
Sorry, did not mean to be pushy. I am just quite unaware of the process and assumed that once the child prs are going to be taken care of once the main one was approved.

Since this one was in drafts for a month, I thought this one got lost. 

Once again sorry for misunderstanding and disturbance!

**Comment by stuartmorgan** [comment]
> I am just quite unaware of the process

We follow the process described at https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins

> Since this one was in drafts for a month, I thought this one got lost.

We don't generally look at draft PRs, we wait for the author of the PR to mark it as ready for review.

File: packages/url_launcher/url_launcher_android/CHANGELOG.md
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_android/CHANGELOG.md:3]: "Adds support for `BrowserConfiguration`."

File: packages/url_launcher/url_launcher_android/example/pubspec.yaml
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_android/example/pubspec.yaml:34]: This should not be needed; the platform interface should be using version constraints at this point.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Review by stuartmorgan — APPROVED** [review]
LGTM, @gmackall for secondary review.

**Review by gmackall — APPROVED** [review]
LGTM!

**Pull request merged**
